src/nvim/eval.lua

```diff
@@ -17,9 +17,10 @@
 --- @field deprecated? true
 --- @field returns? string|false
 --- @field returns_desc? string
+--- @field generics string[] Used to write `---@generics` annotations over a function.
```

Comment:
This is perhaps another hint that suggests we need to migrate to using the meta files as the "source of truth" for the type info instead of re-inventing a type language in `eval.lua`.

Reply:
Keeping the source of truth as a pure structured format makes the generators much simpler overall and better accommodates all the other metadata eval functions have.

I wouldn't say we are re-inventing the type language, it is just being presented in a structured/unwound format.

Reply:
> Keeping the source of truth as a pure structured format makes the generators much simpler overall

? We have the luacats parser, and the meta files are structured.

Reply:
The parser is far from perfect. It's just good enough to build documentation from it. And not requiring a parser is much simpler. It means people can add additional metadata without having to also update a parser.

Reply:
Unrelated but realized to your chat but realized I forgot to add `?` so I changed it to `--- @field generics? string[] ...`